src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -63,20 +72,49 @@ def _process_model_before_weight_loading(self, model, **kwargs):
         from compressed_tensors.quantization import apply_quantization_config
 
         ct_quantization_config = self.compressor.quantization_config
-        apply_quantization_config(model, ct_quantization_config, run_compressed=True)
 
-    def _process_model_after_weight_loading(self, model, **kwargs) -> None:
-        pass
+        # Model is quantized, populate state dict with scale and zp
+        # Compressed models need to be loaded as CompressedLinear for unpacking on decompression
+        apply_quantization_config(
+            model, ct_quantization_config, run_compressed=self.run_compressed or self.is_compressed
+        )
+
+    def _process_model_after_weight_loading(self, model, **kwargs):
+        """Decompress loaded model if necessary - need for qat"""
+
+        if self.is_compressed and not self.run_compressed:
+            config = kwargs.get("config", None)
```

Comment:
If config is required in all cases, it should be defined in the function signature rather than being popped from kwargs

Reply:
we dont need it all cases actually - only for compressed tensors hf quantizer. 
Individual quantizer specific args are in kwargs here